src/ray/raylet/node_manager.cc

```diff
@@ -457,6 +459,17 @@ void NodeManager::Heartbeat() {
         ResourceSet(local_resources.GetLoadResources()));
   }
 
+  // Add resource load by shape. This will be used by the new autoscaler.
+  if (num_heartbeats_before_load_report_ == 0) {
```

Comment:
Doesn't this make aggregation difficult / incorrect if some heartbeats are missing the data? I would prefer to leave this out in the initial version. Also, Ant's delta heartbeat updates might already provide a similar bandwidth saving affect.

Reply:
Ah right, I forgot that we're aggregating across multiple nodes. I'll remove it.